python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,1026 @@
+        doesn't indicate an aperture radius).  For simplicity, do the same for
+        centroids and shapes too.
+
+        Parameters
+        ----------
+        binSize : int
+            Binning factor used.
+        binnedSourceCat : `lsst.afw.table.SourceCatalog`
+            Source catalog from the binned exposure.
+
+        Returns
+        -------
+        table : `astropy.table.Table`
+            Curated source table in unbinned coordinates.
+        """
+        # binSize = binSize
+        table = binnedSourceCat.asAstropy()
+        cols = [n for n in table.colnames if n.startswith("slot")]
+        table = table[cols]
+        if "slot_Centroid_x" in cols:
+            table["slot_Centroid_x"] = (
+                binSize * table["slot_Centroid_x"] + (binSize - 1) / 2
+            )
+            table["slot_Centroid_y"] = (
+                binSize * table["slot_Centroid_y"] + (binSize - 1) / 2
+            )
+        if "slot_Shape_x" in cols:
+            table["slot_Shape_x"] = binSize * table["slot_Shape_x"] + (binSize - 1) / 2
+            table["slot_Shape_y"] = binSize * table["slot_Shape_y"] + (binSize - 1) / 2
+            table["slot_Shape_xx"] *= binSize**2
+            table["slot_Shape_xy"] *= binSize**2
+            table["slot_Shape_yy"] *= binSize**2
+        # area and npixels are just confusing when binning, so remove.
+        if "slot_PsfFlux_area" in cols:
+            del table["slot_PsfFlux_area"]
+        if "slot_PsfFlux_npixels" in cols:
+            del table["slot_PsfFlux_npixels"]
+
+        return table
+
+    def getGoodSources(self, mode, binnedSourceCat):
+        """Perform any filtering on the source catalog
+
+        Parameters
+        ----------
+        mode : {'donut', 'spec', 'photo'}
+            Mode run in.
+        binnedSourceCat : `lsst.afw.table.SourceCatalog`
+            Source catalog from the binned exposure.
+
+        Returns
+        -------
+        wgood : `numpy.ndarray`
+            Boolean array indicating which sources are good.
+        """
+        # Perform any filtering on the source catalog
+        wgood = np.ones(len(binnedSourceCat), dtype=bool)
+        if mode == "spec":
+            fpShapes = [src.getFootprint().getShape() for src in binnedSourceCat]
+            # Filter out likely spectrum detections
+            wgood &= np.array(
+                [sh.getIyy() < 10 * sh.getIxx() for sh in fpShapes], dtype=bool
+            )
+        return wgood
+
+    def getBrightest(self, binnedSourceCat, binSize, wgood):
+        """Find the brightest source in the catalog.
+
+        Parameters
+        ----------
+        binnedSourceCat : `lsst.afw.table.SourceCatalog`
+            Source catalog from the binned exposure.
+        binSize : int
+            Binning factor used.
+        wgood : `numpy.ndarray`
+            Boolean array indicating which sources are good.
+
+        Returns
+        -------
+        maxFluxIdx : int
```

Comment:
And here. Unless this is not necessary for builtins maybe?! If so, I didn't know!